latch_cli/services/register/register.py

```diff
@@ -156,35 +147,49 @@ def _build_and_serialize(
     ctx: _CentromereCtx,
     image_name: str,
     context_path: Path,
-    tmp_dir: Path,
+    tmp_dir: str,
     dockerfile: Optional[Path] = None,
 ):
-    """Builds an image, serializes the workflow within the image, and pushes the image."""
+    assert ctx.pkg_root is not None
 
+    jit_wf = None
     if ctx.workflow_type == WorkflowType.snakemake:
-        wf = extract_snakemake_workflow(ctx.snakefile)
+        assert ctx.snakefile is not None
+        assert ctx.version is not None
+
+        from ...snakemake.serialize import (
+            extract_snakemake_workflow,
+            generate_jit_register_code,
+        )
+
+        wf = extract_snakemake_workflow(ctx.pkg_root, ctx.snakefile)
         jit_wf = wf.build_jit_register_wrapper()
         generate_jit_register_code(
             jit_wf,
             ctx.pkg_root,
             ctx.snakefile,
             ctx.version,
             image_name,
-            user_config.workspace_id,
+            current_workspace(),
         )
 
     image_build_logs = build_image(ctx, image_name, context_path, dockerfile)
     print_and_write_build_logs(image_build_logs, image_name, ctx.pkg_root)
 
     if ctx.workflow_type == WorkflowType.snakemake:
-        serialize_jit_register_workflow(
-            jit_wf, ctx.pkg_root, ctx.snakefile, tmp_dir, image_name, ctx.dkr_repo
-        )
+        assert jit_wf is not None
```

Comment:
i dont understand the asserts, why not check values and throw a more descriptive error, AssertionError tells me nothing (also you can run python without asserts but not as relevant)